src/techui_builder/generate.py

```diff
@@ -59,16 +60,80 @@ def _get_screen_dimensions(self, file: str) -> tuple[int, int]:
 
         return (height, width)
 
+    def _get_screen_dimensions_object(
+        self, object: EmbeddedDisplay | ActionButton
+    ) -> tuple[int, int]:
+        """
+        Parses the bob files for information on the height
+        and width of the screen
+        """
+        # Read the bob file
+        root: etree._Element = etree.fromstring(str(object), None)
+        height_element: etree._Element | None = root.find("height", namespaces=None)
+        if height_element is not None:
+            height = (
+                self.default_size if (val := height_element.text) is None else int(val)
+            )
+        else:
+            height = self.default_size
+            assert "Could not obtain the size of the widget"
+
+        width_element: etree._Element | None = root.find("width", namespaces=None)
+        if width_element is not None:
+            width = (
+                self.default_size if (val := width_element.text) is None else int(val)
+            )
+        else:
+            width = self.default_size
+            assert "Could not obtain the size of the widget"
+
+        return (height, width)
+
+    def _get_screen_position_object(
```

Comment:
As with `_get_screen_dimensions_object`, this should probably omit the "screen" to avoid confusion and avoid using `object` as an argument
```suggestion
    def _get_widget_position(
```